app/depict/src/main/java/org/openscience/cdk/depict/DepictionGenerator.java

```diff
@@ -0,0 +1,667 @@
+    }
+
+    /**
+     * Set the color titles will appear in.
+     *
+     * @param color the color of titles
+     * @return this generator for method chaining
+     * @see RendererModel.TitleColor
+     */
+    public DepictionGenerator withTitleColor(Color color) {
+        return withParam(RendererModel.TitleColor.class,
+                         color);
+    }
+
+    /**
+     * Set the size of titles compared to atom symbols.
+     *
+     * @param scale the scale of titles
+     * @return this generator for method chaining
+     * @see RendererModel.TitleFontScale
+     */
+    public DepictionGenerator withTitleScale(double scale) {
+        return withParam(RendererModel.TitleFontScale.class,
+                         scale);
+    }
+
+    /**
+     * Display atom symbols for terminal carbons (i.e. Methyl)
+     * groups.
+     *
+     * @return this generator for method chaining
+     * @see StandardGenerator.Visibility
+     */
+    public DepictionGenerator withTerminalCarbons() {
+        withParam(StandardGenerator.Visibility.class,
+                  SelectionVisibility.disconnected(SymbolVisibility.iupacRecommendations()));
+        return this;
+    }
+
+    /**
+     * Display atom symbols for all atoms in the molecule.
+     *
+     * @return this generator for method chaining
+     * @see StandardGenerator.Visibility
+     */
+    public DepictionGenerator withCarbonSymbols() {
+        withParam(StandardGenerator.Visibility.class,
+                  SymbolVisibility.all());
+        return this;
+    }
+
+    /**
+     * Highlight the provided set of atoms and bonds in the depiction in the
+     * specified color.
+     * <p/>
+     * Calling this methods appends to the current highlight buffer. The buffer
+     * is cleared after each depiction is generated (e.g. {@link #depict(IAtomContainer)}).
+     *
+     * @param chemObjs set of atoms and bonds
+     * @param color    the color to highlight
+     * @return this generator for method chaining
+     * @see StandardGenerator#HIGHLIGHT_COLOR
+     */
+    public DepictionGenerator withHighlight(Iterable<IChemObject> chemObjs, Color color) {
+        for (IChemObject chemObj : chemObjs)
+            highlight.put(chemObj, color);
+        return this;
+    }
+
+    /**
+     * Specify a desired size of depiction. The units depend on the output format with
+     * raster images using pixels and vector graphics using millimeters. By default depictions
+     * are only ever made smaller if you would also like to make depictions fill all available
+     * space use the {@link #withFillToFit()} option.
+     *
+     * @param w max width
+     * @param h max height
+     * @return this generator for method chaining
+     * @see #withFillToFit()
+     */
+    public DepictionGenerator withSize(double w, double h) {
+        if (w < 0 && h >= 0 || h < 0 && w >= 0)
+            throw new UnsupportedOperationException("Both width and height must be automatic");
```

Comment:
Ah yeah - I'll add a bit there, if it was really cleve we could say make my depictions 128 px high but however wide they need to be (automatic width). Currently it's either specify both width and height or neither.